Web/Scotty/Action.hs

```diff
@@ -101,13 +102,11 @@ runAction :: MonadUnliftIO m =>
           -> ActionT m () -- ^ Route action to be evaluated
           -> m (Maybe Response)
 runAction mh env action = do
-  let
-    handlers = [
-      statusErrorHandler, -- StatusError
-      actionErrorHandler, -- ActionError i.e. Next, Finish, Redirect
-      someExceptionHandler -- all remaining exceptions
-               ]
-  ok <- flip runReaderT env $ runAM $ tryNext (catchesOptionally action mh handlers )
+  ok <- flip runReaderT env $ runAM $ tryNext $ action `catches` concat
+    [ [actionErrorHandler]
+    , maybeToList mh
```

Comment:
good catch!
So going forward: we first handle `Next`, then any user exceptions, then `StatusError` and then everything else. This deserves to be documented properly IMO (see #335 ).